### PR TITLE
Docker container deployment 

### PR DESCRIPTION
**Pull request by mike0sv**

### CODE REVIEW

**Review by aguschin — APPROVED** [review]
LGTM!

One general feedback, the "declare and run" mechanics seems too involved for me each time I try to use it.

For a quicker "I run it and it worked" experience, I would suggest to consider cutting some corners:
1. generate docker image name automatically
2. using FastAPI server automatically
3. allowing to skip environment creation (don't dump it even)
4. allowing to skip deployment declaration creation (don't save it even)

Otherwise you need to read some docs... Which is boresome if you just want to check out how it works. Something like `$ mlem deploy run -m mymodel.mlem -e docker` would work I think.

WDYT? @mike0sv

**Comment by mike0sv** [comment]
1. can be done
2. this will require fastapi dependencies, but might be done
3. hard. we might be able to automatically create env with all defaults, but we still need to dump it probably. or we can treat "no env" as default all the time
4. it can be done now (mlem deploy run can create it for you). not saving will prevent you from managing it after (no status, no teardown, no client requests) so you'll have to do all of this manually which is harder than just save declaration once

**Comment by aguschin** [comment]
Ok, I feel like this should be a part of a bigger discussion, related to https://github.com/iterative/mlem/issues/233. This PR is good to be merged unless you want to add 1 and 2 here.

File: tests/contrib/test_docker/test_deploy.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    DockerContainerState,
+    DockerEnv,
+    DockerImage,
+)
+from mlem.contrib.fastapi import FastAPIServer
+from mlem.core.errors import DeploymentError
+from mlem.core.objects import DeployStatus
+from tests.contrib.test_docker.conftest import docker_test
+
+IMAGE_NAME = "mike0sv/ebaklya"
```
Comment (aguschin) [inline tests/contrib/test_docker/test_deploy.py:20]: Easter egg?

**Review by aguschin — COMMENTED** [review]
🤔

File: tests/contrib/test_docker/test_deploy.py (hunk 1 above)
Comment (mike0sv) [reply tests/contrib/test_docker/test_deploy.py:20]: kinda. it was used by tests before

**Pull request merged**